stdlib/test/builtin/test_reversed.mojo

```diff
@@ -55,29 +71,63 @@ def test_reversed_dict():
 
     assert_equal(keys, "db")
 
+    # got 4 and 2
+    check = 4
+    for val in reversed(dict.values()):
+        assert_equal(val[], check)
+        check -= 2
+
+    keys = String("")
+    check = 4
+    for item in reversed(dict.items()):
+        keys += item[].key
+        assert_equal(item[].value, check)
+        check -= 2
+
+    assert_equal(keys, "db")
+
+    # Refill dict
+    dict["c"] = 2
+    dict["a"] = 1
+    dict["b"] = 4
+    dict["d"] = 3
+
+    # dict: {'b': 4, 'd': 3, 'c': 2, 'a':1}
+
+    keys = String("")
+    check = 1
+    for item in reversed(dict.items()):
+        keys += item[].key
+        assert_equal(item[].value, check)
+        check += 1
+
+    assert_equal(keys, "acdb")
+
     # Empty dict is iterable
 
-    _ = dict.pop("b")
-    _ = dict.pop("d")
+    var empty_dict = Dict[String, Int]()
 
     keys = String("")
-    for key in reversed(dict):
+    for key in reversed(empty_dict):
         keys += key[]
 
     assert_equal(keys, "")
 
-    # Refill dict
+    check = 0
+    for val in reversed(empty_dict.values()):
+        # values is empty, should not reach here
+        check += 1
 
-    dict["d"] = 4
-    dict["a"] = 1
-    dict["b"] = 2
-    dict["e"] = 3
+    assert_equal(check, 0)
```

Comment:
**Question** Perhaps it's easier to just `assert_equal(0, len(empty_dict.values()))`?  Or do you like the extra explicitness here when combined with reverse over the empty values?

Reply:
Yes, I would prefer to test `reversed` with empty dict explicitly. `assert_equal(0, len(empty_dict.values()))` is not testing `reversed` to me